src/openforms/translations/tests/test_api.py

```diff
@@ -0,0 +1,131 @@
+from collections import namedtuple
+
+from django.test import override_settings
+from django.urls import reverse
+from django.utils.translation import gettext as _
+
+from hypothesis import given, settings, strategies as st
+
+# from hypothesis.extra.django import TestCase
+from rest_framework.test import APISimpleTestCase
+
+Language = namedtuple("Language", "conf local")
+
+
+class I18NAPITests(APISimpleTestCase):
+    databases = ["default"]
+
+    @given(
+        st.lists(
+            st.sampled_from(
+                [
+                    Language(("en", _("English")), "English"),
+                    Language(("nl", _("Dutch")), "Nederlands"),
+                    Language(("tr", _("Turkish")), "Türkçe"),
+                ]
+            ),
+            unique=True,
+        )
+    )
+    @settings(deadline=500)
```

Comment:
I've looked a bit more into hypothesis and while I do see the value in it, I'm afraid it's a bit too opaque. Nobody wants to be thinking about or debugging tests to figure out why they are failing, instead, you want to be debugging your actual useful code. We apply the principle that tests should be stupidly easy to follow:

1. data setup
2. call the thing being tested
3. make assertions

This implies:
* avoid putting data/state setup in `setUp` methods or private helper methods -> to understand a failing test, ideally you want to look at only the failing test code without jumping around
* DO repeat yourself - linear straightforward code is easy to follow
* use explicit parametrization if needed and use `self.subTest` with sufficient/relevant context

I find I have to really think here to understand what this test is actually doing, and that hurts maintainability. So please make this a test without hypothesis usage. Instead, we can structure the test more like:

```py
    @override_settings(
        LANGUAGES=(
            ("en", _("English")),
            ("nl-nl", _("Dutch")),
            ("nl-be", _("Flemish")),
        ),
        LANGUAGE_CODE="nl-nl",
    )
    def test_language_info_endpoint(self):
        url = reverse("api:i18n:info")

        response = self.client.get(url)

        expected = {
            "languages": [
                {
                    "code": "en",
                    "name": "English",
                },
                {
                    "code": "nl-nl",
                    "name": "Nederlands",
                },
                {
                    "code": "nl-be",
                    "name": "Vlaams",
                },
            ],
            "current": "nl-nl",
        }
        self.assertEqual(response.json(), expected)
```

This implies:

* fixed order from the settings
* isolated the test from runtime configuration by specifying a _different_ `LANGUAGES` setting and pinning the `LANGUAGE_CODE`

You could also add a smoke test that runs without `@override_settings`, but that's a bit overkill probably.